File: backend/src/YouG.Domain/Entities/EventAttendance.cs
using YouG.Domain.Common;
using YouG.Domain.Enums;

namespace YouG.Domain.Entities;

public class EventAttendance : Entity
{
    public required Guid EventId { get; set; }
    public required Guid UserId { get; set; }
    public required EventAttendanceStatus Status { get; set; }
    public DateTimeOffset RespondedAt { get; set; }
}

File: mobile/lib/features/events/domain/entities/event_attendance.dart
import 'package:freezed_annotation/freezed_annotation.dart';

import 'event_status.dart';

part 'event_attendance.freezed.dart';

@freezed
abstract class EventAttendance with _$EventAttendance {
  const factory EventAttendance({
    required String userId,
    required EventAttendanceStatus status,
    required DateTime respondedAt,
  }) = _EventAttendance;
}
